src/main/java/com/google/devtools/build/lib/remote/RemoteExecutionService.java

```diff
@@ -239,6 +260,11 @@ static Command buildCommand(
       command.setPlatform(platform);
     }
     for (String arg : arguments) {
+      if (actionCompatibleWithXPlatformCache) {
+        // Ensure the command line matches linux for M1
+        arg = arg.replace("macos_aarch64", "linux");
+        arg = arg.replace("darwin_arm64", "linux");
+      }
```

Comment:
Since this is highly use-case specific, I think it would also benefit from being supplied via a flag, e.g. `--remote_xplat_arg_replacements=macos_aarch64=linux,darwin_arm64=linux`).